src/video_core/rasterizer_cache/pixel_format.h

```diff
@@ -29,168 +27,93 @@ enum class PixelFormat : u8 {
-    default:
-        return "NotReal";
-    }
-}
+enum class TextureType : u32 {
+    Texture2D = 0,
+    CubeMap = 1,
+};
 
-constexpr PixelFormat PixelFormatFromTextureFormat(Pica::TexturingRegs::TextureFormat format) {
-    const u32 format_index = static_cast<u32>(format);
-    return (format_index < 14) ? static_cast<PixelFormat>(format) : PixelFormat::Invalid;
-}
+struct PixelFormatInfo {
+    SurfaceType type;
+    std::string_view name;
+    u32 bits_per_block;
+    u32 bytes_per_pixel;
+};
 
-constexpr PixelFormat PixelFormatFromColorFormat(Pica::FramebufferRegs::ColorFormat format) {
-    const u32 format_index = static_cast<u32>(format);
-    return (format_index < 5) ? static_cast<PixelFormat>(format) : PixelFormat::Invalid;
-}
+/**
+ * Lookup table for querying pixel format properties (type, name, etc)
+ * @note Modern GPUs require 4 byte alignment for D24
+ * @note Texture formats are automatically converted to RGBA8
+ **/
+constexpr std::array<PixelFormatInfo, PIXEL_FORMAT_COUNT> FORMAT_MAP = {{
+    {SurfaceType::Color, "RGBA8", 32, 4},  // RGBA8
+    {SurfaceType::Color, "RGB8", 24, 3},   // RGB8
+    {SurfaceType::Color, "RGB5A1", 16, 2}, // RGB5A1
+    {SurfaceType::Color, "RGB565", 16, 2}, // RGB565
+    {SurfaceType::Color, "RGBA4", 16, 2},  // RGBA4
+    {SurfaceType::Texture, "IA8", 16, 4},  // IA8
+    {SurfaceType::Texture, "RG8", 16, 4},  // RG8
+    {SurfaceType::Texture, "I8", 8, 4},    // I8
+    {SurfaceType::Texture, "A8", 8, 4},    // A8
+    {SurfaceType::Texture, "IA4", 8, 4},   // IA4
+    {SurfaceType::Texture, "I4", 4, 4},    // I4
+    {SurfaceType::Texture, "A4", 4, 4},    // A4
+    {SurfaceType::Texture, "ΕTC1", 4, 4},  // ETC1
+    {SurfaceType::Texture, "A4", 8, 4},    // ETC1A4
+    {SurfaceType::Depth, "D16", 16, 2},    // D16
+    {SurfaceType::Invalid, "Invalid", 0},
+    {SurfaceType::Depth, "D24", 24, 4},          // D24
+    {SurfaceType::DepthStencil, "D24S8", 32, 4}, // D24S8
+}};
 
-constexpr PixelFormat PixelFormatFromDepthFormat(Pica::FramebufferRegs::DepthFormat format) {
-    const u32 format_index = static_cast<u32>(format);
-    return (format_index < 4) ? static_cast<PixelFormat>(format_index + 14) : PixelFormat::Invalid;
+constexpr u32 GetFormatBpp(PixelFormat format) {
+    const std::size_t index = static_cast<std::size_t>(format);
+    ASSERT(index < FORMAT_MAP.size());
+    return FORMAT_MAP[index].bits_per_block;
 }
 
-constexpr PixelFormat PixelFormatFromGPUPixelFormat(GPU::Regs::PixelFormat format) {
-    const u32 format_index = static_cast<u32>(format);
-    switch (format) {
-    // RGB565 and RGB5A1 are switched in PixelFormat compared to ColorFormat
-    case GPU::Regs::PixelFormat::RGB565:
-        return PixelFormat::RGB565;
-    case GPU::Regs::PixelFormat::RGB5A1:
-        return PixelFormat::RGB5A1;
-    default:
-        return (format_index < 5) ? static_cast<PixelFormat>(format) : PixelFormat::Invalid;
-    }
+constexpr u32 GetBytesPerPixel(PixelFormat format) {
```

Comment:
```suggestion
constexpr u32 GetFormatBytesPerPixel(PixelFormat format) {
```
to make it consistent with the other functions